### PR TITLE
Get latitude and longitude on Business model creation

### PR DESCRIPTION
**Pull request by rschalch**
added method to, given an address, save lat and lng in the Business model

### CODE REVIEW

File: my_community/core/models.py [hunk 1]
```diff
@@ -33,3 +35,26 @@ class Business(models.Model):
...
+            address = address
+
+        gmaps = gm.Client(key=config('GOOGLE_API_KEY'))
+        geocode = gmaps.geocode(address)
+        lat = geocode[0]['geometry']['location']['lat']
+        lng = geocode[0]['geometry']['location']['lng']
+        return (lat, lng)
+
+    def save(self, *args, **kwargs):
+        self.latitude = self.get_lat_lng(self.address)[0]
```
Comment (cacarrara) [inline my_community/core/models.py]: We do not need to pass an attribute as argument to a method, I mean, `self.address` is accessible inside method `get_lat_lng`.

File: my_community/core/models.py [hunk 2]
```diff
@@ -33,3 +35,26 @@ class Business(models.Model):
...
+        """
+        Description: given a certain address, get latitude and longitude
+        Returns: a tuple containing [0] latitude and [1] longitude
+        """
+        if address is None:
+            address = self.address
+        else:
+            address = address
+
+        gmaps = gm.Client(key=config('GOOGLE_API_KEY'))
```
Comment (cacarrara) [inline my_community/core/models.py]: `GOOGLE_API_CONFIG` must be a setting defined in `settings.py`

File: my_community/core/models.py [hunk 3]
```diff
@@ -33,3 +35,26 @@ class Business(models.Model):
 
     def __str__(self):
         return "{} ({} - {})".format(self.name, self.segment, self.business_type)
+
+    def get_lat_lng(self, address=None):
+        """
+        Description: given a certain address, get latitude and longitude
```
Comment (cacarrara) [inline my_community/core/models.py]: I think this comment is unnecessary

File: my_community/core/models.py [hunk 4]
```diff
@@ -1,5 +1,7 @@
 from django.db import models
 from django.utils.translation import ugettext_lazy as _
+from decouple import config
+import googlemaps as gm
```
Comment (cacarrara) [inline my_community/core/models.py]: I think renaming this module is unnecessary and less explicity.

File: my_community/core/models.py [hunk 5]
```diff
@@ -33,3 +35,26 @@ class Business(models.Model):
...
+        Returns: a tuple containing [0] latitude and [1] longitude
+        """
+        if address is None:
+            address = self.address
+        else:
+            address = address
+
+        gmaps = gm.Client(key=config('GOOGLE_API_KEY'))
+        geocode = gmaps.geocode(address)
+        lat = geocode[0]['geometry']['location']['lat']
```
Comment (cacarrara) [inline my_community/core/models.py]: If I'm not wrong we can access lat and ln this way: `geocode[0]['location']['lat']`. Also, what if maps api do not found any address? Did you make sure that 0 index is there?

File: my_community/core/models.py [hunk 6]
```diff
@@ -33,3 +35,26 @@ class Business(models.Model):
 
     def __str__(self):
         return "{} ({} - {})".format(self.name, self.segment, self.business_type)
+
+    def get_lat_lng(self, address=None):
```
Comment (cacarrara) [inline my_community/core/models.py]: Where is the tests for it?

File: my_community/core/models.py [hunk 7]
```diff
@@ -33,3 +35,26 @@ class Business(models.Model):
...
+        lat = geocode[0]['geometry']['location']['lat']
+        lng = geocode[0]['geometry']['location']['lng']
+        return (lat, lng)
+
+    def save(self, *args, **kwargs):
+        self.latitude = self.get_lat_lng(self.address)[0]
+        self.longitude = self.get_lat_lng(self.address)[1]
+
+        # ensure the save method is called
+        super(Business, self).save(*args, **kwargs)
```
Comment (cacarrara) [inline my_community/core/models.py]: We must return the save result: `return super(Business, self).save(* args, **kwargs)`

File: my_community/core/models.py [hunk 8]
```diff
@@ -33,3 +35,26 @@ class Business(models.Model):
...
+        if address is None:
+            address = self.address
+        else:
+            address = address
+
+        gmaps = gm.Client(key=config('GOOGLE_API_KEY'))
+        geocode = gmaps.geocode(address)
+        lat = geocode[0]['geometry']['location']['lat']
+        lng = geocode[0]['geometry']['location']['lng']
+        return (lat, lng)
```
Comment (cacarrara) [inline my_community/core/models.py]: Why return lat and lng if we can directly put them in proper attributes?

File: my_community/core/models.py (hunk 5 above)
Comment (rschalch) [inline my_community/core/models.py]: I tested this, we need geomtry

File: my_community/core/models.py [hunk 9]
```diff
@@ -33,3 +35,26 @@ class Business(models.Model):
...
         return "{} ({} - {})".format(self.name, self.segment, self.business_type)
+
+    def get_lat_lng(self, address=None):
+        """
+        Description: given a certain address, get latitude and longitude
+        Returns: a tuple containing [0] latitude and [1] longitude
+        """
+        if address is None:
+            address = self.address
+        else:
```
Comment (lucasmagnum-luizalabs) [inline my_community/core/models.py]: This `else` is unecessary,  when `address` is not None, the `address` variable  is already assigned.

File: my_community/core/models.py (hunk 4 above)
Comment (cacarrara) [inline my_community/core/models.py]: Also, we need to add googlemaps to requirements file.

**Pull request closed without merging**